experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -463,3 +500,130 @@ function Checkpointer.get_model_prog_state(sim::ClimaLandSimulation)
+
+    # update `csf` in-place using this model's flux calculation
+    fluxes = coupler_land_turbulent_fluxes!(p, sim.model, Y, t, coupled_atmos)
+    (; F_turb_ρτxz, F_turb_ρτyz, F_shf, F_lhf, F_turb_moisture) = fluxes
+
+    # add the flux contributing from this surface to the coupler field
+    # note that the fluxes are area-weighted, so if a surface model is
+    #  not present at this point, the fluxes are zero
+    @. csf.F_turb_ρτxz += F_turb_ρτxz * area_fraction * area_mask
+    @. csf.F_turb_ρτyz += F_turb_ρτyz * area_fraction * area_mask
+    @. csf.F_turb_energy += (F_shf .+ F_lhf) * area_fraction * area_mask
+    @. csf.F_turb_moisture += F_turb_moisture * area_fraction * area_mask
+    return nothing
+end
+
+"""
+    coupler_land_turbulent_fluxes(model, Y, p, t, coupled_atmos)
+
+Compute the turbulent fluxes for the integrated land model by computing them
+for each subcomponent, then combining them.
+It takes in 4 arguments related to the land model simulation: `model`, `Y`, `p`, and `t`.
+It also takes in `coupled_atmos`, a ClimaLand.CoupledAtmosphere containing the
+atmospheric properties required for the flux calculation: `u`, `h`,
+`thermal_state`, and `gustiness`.
+
+This function updates each subcomponent in place with its own turbulent fluxes, and also
+returns the total turbulent fluxes for the land model.
+
+Note that this function is currently allocating. We should rewrite it/allocate
+space so that it can update fields in-place instead.
+"""
+function coupler_land_turbulent_fluxes!(p, model::ClimaLand.LandModel, Y, t, coupled_atmos::ClimaLand.CoupledAtmosphere)
+    # compute the fluxes for each sub-component and update the land model cache
+    soil_dest = p.soil.turbulent_fluxes
+    ClimaLand.coupler_compute_turbulent_fluxes!(soil_dest, coupled_atmos, model.soil, Y, p, t)
+
+    snow_dest = p.snow.turbulent_fluxes
+    ClimaLand.coupler_compute_turbulent_fluxes!(snow_dest, coupled_atmos, model.snow, Y, p, t)
+
+    canopy_dest = p.canopy.turbulent_fluxes
+    ClimaLand.coupler_compute_turbulent_fluxes!(canopy_dest, coupled_atmos, model.canopy, Y, p, t)
+
+    # combine fluxes from each component of the land model
+    F_lhf = @. canopy_dest.lhf +
+       soil_dest.lhf * (1 - p.snow.snow_cover_fraction) +
+       p.snow.snow_cover_fraction * snow_dest.lhf
+    F_shf = @. canopy_dest.shf +
+       soil_dest.shf * (1 - p.snow.snow_cover_fraction) +
+       p.snow.snow_cover_fraction * snow_dest.shf
+    F_turb_moisture = @. canopy_dest.transpiration +
+       (soil_dest.vapor_flux_liq + soil_dest.vapor_flux_ice) * (1 - p.snow.snow_cover_fraction) +
+       p.snow.snow_cover_fraction * snow_dest.vapor_flux
+    F_turb_ρτxz = @. soil_dest.ρτxz * (1 - p.snow.snow_cover_fraction) + p.snow.snow_cover_fraction * snow_dest.ρτxz
+    F_turb_ρτyz = @. soil_dest.ρτyz * (1 - p.snow.snow_cover_fraction) + p.snow.snow_cover_fraction * snow_dest.ρτyz
+
+    # At locations where this surface model is not evaluated, we get `NaN` for
```

Comment:
Is it correct to say that the model is not evaluated, and are these really NaN? Or are they computed but irrelevant/wrong because over ocean? 

Also, for CL right now, I think we may have some NaNs over the ocean, but also other "physical" but meaningless values. This would just replace the NaN values, leaving some ocean values set, right?

Reply:
These lines also have the affect of hiding NaNs that appear over land (setting them to 0).